refactor: cuesheet time format

Shobhit-Nagpal opened the pull request:
## Changes made:
- added `timeFormat` as a type under Tanstack table's declaration
- get `timeFormat` with `useSettings` from `<CuesheetTable>`
- add `timeFormat` as an optional prop under `<TimeInput>` that is under cuesheet-elements
- add `timeFormat` as an option under `FormatOptions` and change the hour formatting based on that inside `millisToString` util

Issue: There's an issue currently with the Duration column in the table. If we toggle the focus between `timeEnd` and `duration` column, the formatting of the duration seems to add an additional '12h'

For example: if timeStart is 1:05:00 and timeEnd is 1:25:00, and we toggle focus, the duration changes from '20m' to '12h20m'

Shobhit-Nagpal commented:
Hey, made a couple changes, sorted out the noted issue as well.

Changes:
- switched `millisToString` to `formatTime` under `cuesheetCols.tsx` for timeStart and timeEnd
- handled case for `hideTableSeconds` if time format is 12h
- changed the `timeFormat` prop to a `isFormat` boolean (this is to differentiate formatting of duration vs timeEnd and timeStart while editing the field)

Attaching a video to show the working:

https://github.com/user-attachments/assets/525de903-4c27-48c6-ae0d-3f0bc1255c4c

Do let me know if there are any changes required

cpvalente commented on apps/client/src/views/cuesheet/cuesheet-table/cuesheet-table-elements/TimeInput.tsx:
```diff
@@ -18,7 +21,7 @@ interface ParentFocusableInput extends HTMLInputElement {
 export default memo(TimeInputDuration);
 
 function TimeInputDuration(props: PropsWithChildren<TimeInputDurationProps>) {
```
I see, I would suggest creating a separate `<DurationInput>` so we can make more assumptions about what data is in there

Shobhit-Nagpal replied:
wouldn't that be repeating most of the code already present with `<TimeInput>`?

it makes sense to be more specific with the component but that would be potentially duplicating code i believe

cpvalente commented on apps/client/src/views/cuesheet/cuesheet-table/cuesheet-table-elements/cuesheetCols.tsx:
```diff
@@ -26,13 +26,18 @@ function MakeStart({ getValue, row, table }: CellContext<OntimeEntry, unknown>)
   const displayTime = showDelayedTimes ? startTime + delayValue : startTime;
-  let formattedTime = millisToString(displayTime);
+  let formattedTime = formatTime(displayTime)
   if (hideTableSeconds) {
-    formattedTime = removeSeconds(formattedTime);
+    if (timeFormat === '12') {
+      const [parsingTime, amPmIndicator] = formattedTime.split(' ');
+      formattedTime = removeSeconds(parsingTime) + ' ' + amPmIndicator
+    } else {
+      formattedTime = removeSeconds(formattedTime);
+    }
```
I see, could we pass the formats to the formatTime function?

ie: if user says remove seconds, we pass `hh:mm` a otherwise `hh:mm:ss a`?

Shobhit-Nagpal commented:
## Changes made on recent push:

- added a `<DurationInput>` which is being used for duration field under `MakeDuration`
- removed `isFormat` prop from `<TimeInput>` and the component is now being used for timeEnd and timeStart
- refactored logic of showing seconds by passing format options to `formatTime`
- removed passing of `timeFormat` as option under `<CuesheetTable` as we get the time format from settings in `formatTime`, originally `timeFormat` was being passed to handle logic of viewing/hiding seconds which was resolved from the above point, hence removed this
- removed `timeFormat` under the declaration file
